Tasks/AzureAppServiceManage/appinsightsmanage.ts

```diff
@@ -0,0 +1,163 @@
+import tl = require('vsts-task-lib/task');
+import Q = require('q');
+import path = require('path');
+
+var azureRESTUtils = require('azurerest-common/azurerestutility.js');
+var azureUtils = require('azurerest-common/utility.js');
+var extensionManage = require('./extensionmanage.js');
+
+var APPLICATION_INSIGHTS_EXTENSION_NAME = "Microsoft.ApplicationInsights.AzureWebSites";
+
+export class AppInsightsManage {
+
+    private endpoint;
+    private appInsightsResourceGroupName: string;
+    private appInsightsResourceName: string;
+    private webAppName: string;
+    private resourceGroupName: string;
+    private specifySlotFlag: boolean;
+    private slotName: string;
+
+    constructor(endpoint, appInsightsResourceGroupName, appInsightsResourceName, webAppName, resourceGroupName, specifySlotFlag, slotName) {
+        this.endpoint = endpoint;
+        this.appInsightsResourceGroupName = appInsightsResourceGroupName;
+        this.appInsightsResourceName = appInsightsResourceName;
+        this.webAppName = webAppName;
+        this.resourceGroupName = resourceGroupName;
+        this.specifySlotFlag = specifySlotFlag;
+        this.slotName = slotName;
+    }
+
+    public async configureAppInsights() {
+        this.resourceGroupName  = (this.specifySlotFlag ? this.resourceGroupName : await azureRESTUtils.getResourceGroupName(this.endpoint, this.webAppName));
```

Comment:
I get that resourcegroup name input is available when slot is specified. But from code point of view, can't you check if resourcegroupname is null or empty to decide if you need to make a rest call?